web/src/typeahead_helper.ts

```diff
@@ -523,11 +514,13 @@ export function sort_recipients<UserType extends UserOrMentionPillData | UserPil
 
     function add_user_recipients(items: UserType[]): void {
         for (const item of items) {
-            const item_is_broadcast = item.type === "user_or_mention" && item.is_broadcast;
-            const topic_wildcard_mention = item.email === "topic";
-            if (!item_is_broadcast || topic_wildcard_mention || !stream_wildcard_mention_included) {
+            if (
+                item.type !== "broadcast" ||
+                item.user.email === "topic" ||
+                !stream_wildcard_mention_included
+            ) {
                 recipients.push(item);
-                if (item_is_broadcast && !topic_wildcard_mention) {
+                if (item.type === "broadcast" && item.user.email !== "topic") {
```

Comment:
I guess the next sanity-repairing changes would be to change the string for a broadcast mention to no longer be called `.email` .